### PR TITLE
Feature/explorer tests

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
Explorer related tests added

### CODE REVIEW

File: app/client/cypress/locators/DatasourcesEditor.json [hunk 1]
```diff
@@ -13,5 +13,6 @@
   "PostgreSQL": ".t--plugin-name:contains('PostgreSQL')",
   "sectionAuthentication": "[data-cy=section-Authentication]",
   "sectionSSL": "[data-cy=section-SSL\\ \\(optional\\)]",
-  "addDatasourceEntity": "//div[contains(@class,'t--entity group plugins')]//div[contains(@class,'t--entity-add-btn')]"
+  "addDatasourceEntity": "//div[contains(@class,'t--entity group plugins')]//div[contains(@class,'t--entity-add-btn')]",
```
Comment (hetunandu) [inline app/client/cypress/locators/DatasourcesEditor.json:16]: Can we create a better selector for this button?

File: app/client/cypress/locators/apiWidgetslocator.json [hunk 2]
```diff
@@ -42,5 +42,8 @@
   "requestBody": "(//div[contains(@class,'bp3-collapse-body')]//textarea)[1]",
   "showrequest": "span:contains('Show Request')",
   "Responsetab": "//li[text()='Response Body']",
-  "deleteAPI": ".t--apiFormDeleteBtn"
+  "deleteAPI": ".t--apiFormDeleteBtn",
+  "editName": ".single-select >div:contains('Edit Name')",
+  "page": ".single-select >div:contains('",
```
Comment (hetunandu) [inline app/client/cypress/locators/apiWidgetslocator.json:47]: This should not be a string but a function that creates this selector with the page name in the right place

File: app/client/cypress/locators/explorerlocators.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+    "NoApiMsg": "p:contains('No APIs yet.')",
+    "NoQueryMsg": "p:contains('No Queries yet.')",
+    "NoWidgetsMsg": "p:contains('No widgets yet.')",
+    "AddPage": "//div[contains(@class,'t--entity group pages')]//div[contains(@class,'t--entity-add-btn')]",
```
Comment (hetunandu) [inline app/client/cypress/locators/explorerlocators.json:5]: Update selector

File: app/client/cypress/locators/explorerlocators.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    "addEntityAPI": "//div[contains(@class,'t--entity group apis')]//div[contains(@class,'t--entity-add-btn')]",
```
Comment (hetunandu) [inline app/client/cypress/locators/explorerlocators.json:6]: Update selector

File: app/client/cypress/support/commands.js [hunk 5]
```diff
@@ -367,6 +368,17 @@ Cypress.Commands.add("EditApiName", apiname => {
   cy.WaitAutoSave();
 });
 
+Cypress.Commands.add("EditApiNameFromExplorer", apiname => {
+  cy.xpath(apiwidget.popover)
+    .last()
+    .click({ force: true });
+  cy.get(apiwidget.editName).click({ force: true });
+  cy.xpath('//div[contains(@class,"t--entity-name")]/input')
```
Comment (hetunandu) [inline app/client/cypress/support/commands.js:376]: This was defined in the selector. Avoid using xpath selectors

File: app/client/cypress/support/commands.js [hunk 6]
```diff
@@ -607,17 +640,30 @@ Cypress.Commands.add("CopyAPIToHome", () => {
...
+  cy.deleteAction();
   cy.wait("@deleteAction").should(
     "have.nested.property",
     "response.body.responseMeta.status",
     200,
   );
 });
 
+Cypress.Commands.add("DeleteWidgetFromSideBar", () => {
+  cy.deleteAction();
```
Comment (hetunandu) [inline app/client/cypress/support/commands.js:652]: The command is called delete action but being used to delete a widget

File: app/client/cypress/locators/DatasourcesEditor.json (hunk 1 above)
Comment (NandanAnantharamu) [reply app/client/cypress/locators/DatasourcesEditor.json]: done

File: app/client/cypress/locators/explorerlocators.json (hunk 3 above)
Comment (NandanAnantharamu) [reply app/client/cypress/locators/explorerlocators.json]: done

File: app/client/cypress/locators/explorerlocators.json (hunk 4 above)
Comment (NandanAnantharamu) [reply app/client/cypress/locators/explorerlocators.json]: done

File: app/client/cypress/support/commands.js (hunk 6 above)
Comment (NandanAnantharamu) [reply app/client/cypress/support/commands.js]: done

**Review by hetunandu — APPROVED** [review]

**Pull request merged**